src/sonic-bgpcfgd/bgpcfgd/manager_static_route.py

```diff
@@ -0,0 +1,173 @@
+import traceback
+from .log import log_crit, log_err, log_debug
+from .manager import Manager
+from .template import TemplateFabric
+import socket
+
+class StaticRouteMgr(Manager):
+    """ This class updates static routes when STATIC_ROUTE table is updated """
+    def __init__(self, common_objs, db, table):
+        """
+        Initialize the object
+        :param common_objs: common object dictionary
+        :param db: name of the db
+        :param table: name of the table in the db
+        """
+        super(StaticRouteMgr, self).__init__(
+            common_objs,
+            [],
+            db,
+            table,
+        )
+
+        self.static_routes = {}
+
+    OP_DELETE = 'DELETE'
+    OP_ADD = 'ADD'
+
+    def set_handler(self, key, data):
+        vrf, ip_prefix = self.split_key(key)
+        is_ipv6 = TemplateFabric.is_ipv6(ip_prefix)
+
+        arg_list    = lambda v: v.split(',') if len(v.strip()) != 0 else None
+        bkh_list    = arg_list(data['blackhole']) if 'blackhole' in data else None
+        nh_list     = arg_list(data['nexthop']) if 'nexthop' in data else None
+        intf_list   = arg_list(data['ifname']) if 'ifname' in data else None
+        dist_list   = arg_list(data['distance']) if 'distance' in data else None
+        nh_vrf_list = arg_list(data['nexthop-vrf']) if 'nexthop-vrf' in data else None
+
+        try:
+            ip_nh_set = IpNextHopSet(is_ipv6, bkh_list, nh_list, intf_list, dist_list, nh_vrf_list)
+            cur_nh_set = self.static_routes.get(vrf, {}).get(ip_prefix, IpNextHopSet(is_ipv6))
+            cmd_list = self.static_route_commands(ip_nh_set, cur_nh_set, ip_prefix, vrf)
+        except Exception as exc:
+            log_crit("Got an exception %s: Traceback: %s" % (str(exc), traceback.format_exc()))
+            return False
+
+        if cmd_list:
+            self.cfg_mgr.push_list(cmd_list)
+            log_debug("Static route {} is scheduled for updates".format(key))
+        else:
+            log_debug("Nothing to update for static route {}".format(key))
+
+        self.static_routes.setdefault(vrf, {})[ip_prefix] = ip_nh_set
+
+        return True
+
+
+    def del_handler(self, key):
+        vrf, ip_prefix = self.split_key(key)
+        is_ipv6 = TemplateFabric.is_ipv6(ip_prefix)
+
+        ip_nh_set = IpNextHopSet(is_ipv6)
+        cur_nh_set = self.static_routes.get(vrf, {}).get(ip_prefix, IpNextHopSet(is_ipv6))
+        cmd_list = self.static_route_commands(ip_nh_set, cur_nh_set, ip_prefix, vrf)
+
+        if cmd_list:
+            self.cfg_mgr.push_list(cmd_list)
```

Comment:
Can we add test cases with other L3 interfaces as well e.g PHY/VLAN..etc.

Reply:
I am not sure if I fully understood your suggestion. Could you please maybe give an example of what additional test case you are suggesting?

Reply:
Looks like you have added the test cases with Ethernet already, thanks.